### PR TITLE
use restricted-scc in provider deployment

### PR DESCRIPTION
**Pull request by cheewaio**
### Description of your changes

Fixes #

Removes runAsUser and runAsGroup in provider deployment, so the pod can be run with restricted scc in OpenShift

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: internal/controller/pkg/revision/deployment.go [hunk 1]
```diff
@@ -28,9 +28,10 @@ import (
 )
 
 var (
-	replicas                 = int32(1)
-	runAsUser                = int64(2000)
-	runAsGroup               = int64(2000)
+	replicas = int32(1)
+	// IBM Patch: Disable to run with restricted-scc in OpenShift
```
Comment (bitscuit) [inline internal/controller/pkg/revision/deployment.go:32]: should we comment it out? Or just remove the lines completely and then write a good commit message to describe the changes, e.g.:

> removed "runAsUser" and "runAsGroup" variables from deployment, so can run with restricted-scc in OpenShift

Comment (cheewaio) [reply internal/controller/pkg/revision/deployment.go:32]: I wanted to be able to quickly identify which files are modified by IBM by searching for `IBM Patch`

Comment (bitscuit) [reply internal/controller/pkg/revision/deployment.go:32]: sounds reasonable. For situations where upstream decides to remove those lines, we should remove them as well when syncing the fork.

File: common/Makefile.common.mk [hunk 2]
```diff
@@ -25,6 +25,7 @@
 BUILD_LOCALLY ?= 1
 IMAGE_NAME ?= ibm-crossplane
 RELEASE_VERSION ?= $(shell cat RELEASE_VERSION)
+GO_SUPPORTED_VERSIONS = 1.14|1.15
```
Comment (bitscuit) [inline common/Makefile.common.mk:28]: is this necessary? Golang is strict with backwards compatibility, so if go 1.14 works, then go 1.15 is implied to also work

Comment (cheewaio) [reply common/Makefile.common.mk:28]: crossplane build script only supports up to 1.14 .. the variable is to override crossplane build script to use go 1.15

Comment (cheewaio) [reply common/Makefile.common.mk:28]: the whole common directory is added by IBM

**Review by bitscuit — APPROVED** [review]
/lgtm

**Pull request merged**